### ISSUE TITLE
Bug: Summaries of asset created and updated datetimes breaks the stac spec

### ISSUE THREAD

**Issue by billgeo** [linz/stac#143]
[Please remove this and other [placeholder text] before submitting.]

**Describe the bug**
Currently the summaries we have specified look like this: https://github.com/linz/stac/blob/7443877bda35b5b519cb213c3b5dfb20d351a73d/extensions/linz/examples/collection.json#L39-L51

This is what the summary of the item created updated datetimes look like (when the metadata is updated, not the asset).

See discussion here. https://github.com/radiantearth/stac-spec/issues/1153 and https://github.com/radiantearth/stac-spec/discussions/1156

Could look like this:

```
   "summaries": {
      "created": {
        "minimum": "1901-01-01T00:00:00Z",
        "maximum": "1920-01-01T00:00:00Z"
      },
      "updated": {
        "minimum": "1901-01-02T00:00:00Z",
        "maximum": "1920-01-02T00:00:00Z"
      }
  }
```
    

And the the summary with other `properties` would look something like this.

```
   "summaries": {
    "assets": {
      "created": {
        "minimum": "1901-01-01T00:00:00Z",
        "maximum": "1920-01-01T00:00:00Z"
      },
      "updated": {
        "minimum": "1901-01-02T00:00:00Z",
        "maximum": "1920-01-02T00:00:00Z"
      }
    },
    "platform": ["Fixed-wing Aircraft"],
    "instruments": ["EAGLE IV"],
    "created": {
      "minimum": "1999-01-01T00:00:00Z",
      "maximum": "2010-01-01T00:00:00Z"
    },
    "updated": {
      "minimum": "1999-01-02T00:00:00Z",
      "maximum": "2010-01-02T00:00:00Z"
    } 
  },
```

**To Reproduce**

**Expected behavior**
[A clear and concise description of what you expected to happen.]

**Actual behavior**
[Describe what actually happened.]

[If applicable, add logs to help explain your problem:]

```console
[Paste logs here]
```

**Additional context**
[Add any other context about the problem here.]

#### Tasks

- [x] Check if summaries can be extended or not
- [ ] If we can, create new summary properties that relate only to assets and make mandatory
- [x] If we can't, discuss alternatives to capturing this metadata somewhere and make mandatory
- [x] Remove mandatory schema check for summaries having an 'updated` and `created` datetimes [Bill confirmed this is [only mandatory for published data](https://confluence.linz.govt.nz/display/LIFFP/STAC+Field+Mapping#STACFieldMapping-CoreDatasetMetadataFields) so the mandatory schema check can be removed]

**Comment by l0b0** [comment]
Blocked by https://github.com/linz/stac/pull/157#issuecomment-968406114.

**Comment by billgeo** [comment]
This is done now. Will open a new issue if there are changes/recommendations from upstream.
